lib/stylesheet/manager/builder.rb

```diff
@@ -36,7 +36,7 @@ def compile(opts = {})
          rtl: rtl,
          theme_id: theme&.id,
          theme_variables: theme&.scss_variables.to_s,
-         source_map_file: source_map_filename,
+         source_map_file: source_map_url_relative_from_stylesheet,
```

Comment:
This is better, but I still find it a bit strange that we assign a relative URL to a filename. I don't know what "right" would look like, so I think you can leave it this way.